File: trunk/quote/Projects/PluginModel/ModelEvents.vb

Imports System.ComponentModel

Imports Model.Common

Public Class ModelEventArgs
    Public Property ID
End Class

Public Class ModelEvents

    Public Shared ReadOnly ActiveEvents As ModelEvents = New ModelEvents

    Public Delegate Sub ModelEventHandler(source As Object, args As ModelEventArgs)

    Public Shared Event TemplateCreated As ModelEventHandler

    Public Shared Sub NotifyTemplateCreated(id As Integer)
        Dim args As New ModelEventArgs
        args.ID = id
        RaiseEvent TemplateCreated(Nothing, args)
    End Sub

End Class

File: trunk/quote/Projects/PluginModel/ModelEvents.cs
namespace Model
{
    using System;
    using System.Linq;
    
    public sealed class ModelEvents
    {
        private static ModelEvents instance = new ModelEvents();
        
        private ModelEvents()
        {
            // Singleton class
        }

        public event EventHandler<ModelEventArgs> TemplateCreated;

        public event EventHandler TemplateViewed;

        public event EventHandler QuoteViewed;

        public static ModelEvents Instance
        {
            get { return instance; }
        }

        public void NotifyTemplateCreated(int id)
        {
            ModelEventArgs args = new ModelEventArgs();
            args.Id = id;
            if (this.TemplateCreated != null) 
            {
                this.TemplateCreated(null, args);
            }
        }

        public void NotifyTemplateViewed()
        {
            if (this.TemplateViewed != null) 
            {
                this.TemplateViewed(null, new EventArgs());
            }
        }

        public void NotifyQuoteViewed()
        {
            if (this.QuoteViewed != null) 
            {
                this.QuoteViewed(null, new EventArgs());
            }
        }
    }
}